### ISSUE TITLE
Circle CI darwin break

### ISSUE THREAD

**Issue by guibou** [tweag/rules_haskell#450]
Since a few time, most CI jobs on darwin fails with "empty" archives. Even commits which were previously working, that's why I'm suspecting a change of configuration of the darwin worker.

`ar` seems unable to generate correct archives and returns a 0 bytes file plus a warning:

```
/bin/tests/visible-binaries/xcrunwrapper.sh: line 40: SDKROOT: unbound variable
```

Using one of the `Path` from `xcodebuild -sdk -version`, such as:

```
export SDKROOT=/Applications/Xcode-9.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.0.sdk
```

fixes the build if I'm calling the command manually (i.e. using the output of `bazel -s`). I'm unable to reproduce this fix using any of the environment variable passing method I know for bazel (e.g. using `action_env=SDKROOT=...` for example). Apparently the `xrunwrapper.sh` replaces some constant by `SDKROOT`, but they may be provided by bazel.

My knowledge of the darwin ecosystem is limited (polite way of saying that I know nothing), so I'd like your help to decide if we should hard code this variable in the circle CI configuration file or if another approach is recommended.

@judah who wrote the original code involving `xrunwrapper.sh`.

**Comment by guibou** [comment]
I set this as high priority because broken CI is bad, especially that it introduces false negative which may confuse devs. For example #448 seems perfectly fine but fails on CI.

**Comment by guibou** [comment]
It may be an upgrade to bazel which cause the problem, currently our circle CI build is not pinned to a fixed bazel version.
